obspy/io/nied/util.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+from obspy import UTCDateTime
+
+def gen_sc3_id(dt, numenc=6, sym="abcdefghijklmnopqrstuvwxyz"):
+    """
+    Generate an event ID following the SeisComP3 convention. By default it
+    divides a year into 26^6 intervals assigning each a unique combination of
+    characters.
+
+    >>> print(gen_sc3_id(UTCDateTime(2015, 8, 18, 10, 55, 51, 367580)))
+    2015qffasl
+    """
+    numsym = len(sym)
+    x = (((((dt.julday - 1) * 24) + dt.hour) * 60 + dt.minute) * 60 + dt.second) * 1000 \
+    + dt.microsecond / 1000
+    dx = (((370 * 24) * 60) * 60) * 1000
+    rng = 1
+    tmp = rng
+    for i in xrange(numenc):
```

Comment:
`xrange` does not exist on Python 3. The `from future.builtins import *` line actually make the normal `range()` behave like `xrange()` on Python 2 so just using `range` is always ok.